docs/schema.md

```diff
@@ -921,6 +921,68 @@ If this section is not specified, users see a warning when deploying the app to
 an Istio-enabled environment. The [`ISTIO_ENABLED`](#type-istio_enabled) type
 indicates whether Istio is enabled on the cluster.
 
+### assistedClusterCreation
+
+Use `assistedClusterCreation` to specify constraints for cluster creation as part of app deployment. These constraints determine whether a new cluster creation is allowed as part of deployment; and if yes, what are the node constraints.
+
+```yaml
+properties:
+  # Property definitions...
+required:
+  # Required properties...
+x-google-marketplace:
+  clusterConstraints:
+    assistedClusterCreation:
+      type: DISABLED | STRICT
+      creationGuidance: #string
+      gke:
+        nodePool:
+        - numNodes: #integer
+          machineType: #string
+```
+
+Supported types:
+- `DISABLED`: The app doesn't support creating a new cluster as part of app deployment. User must select a pre-existing cluster.
```

Comment:
Note here or in the `creationGuidance` explanatory line that `DISABLED` requires `creationGuidance`. We are fine with partners providing it when enabled, but we don't currently do anything with it and it is a bit awkward since Marketplace creates the cluster for the user (though we concieveably could show that info to help the user create their own customized cluster that will still work with the app)